### ISSUE TITLE
grdcut -D downloads remote files on linux and windows

### ISSUE THREAD

**Issue by meghanrjones** [GenericMappingTools/gmt#5809]
**Description of the problem**

grdcut -D should only report information, not download any files. Recent CI failures show that on windows and linux, grdcut -D attempts to download the file at the determined resolution.

**Full script that generated the error**

```
gmt set FORMAT_FLOAT_OUT %.5lf
PROJ=-JG15/20/2500/0/0/0/30/30/15c
gmt grdcut @earth_age    -Rg $PROJ -D -V >> result.txt
```

**Full error message**

```
[NOTICE]: Remote data courtesy of GMT data server oceania [http://oceania.generic-mapping-tools.org]
[NOTICE]: Earth Seafloor Crustal Age (GTS12 timescale) at 1x1 arc minutes from EarthByte [Seton et al., 2020].
[NOTICE]:   -> Download grid file [188M]: earth_age_01m_g/
```

**System information**

* Operating system: ubuntu 20.04
* GMT version (`gmt --version`): 6.3.0_9ff054a-dirty_2021.09.27

**Comment by PaulWessel** [comment]
Thanks, I think I understand why.  The determination of the resolution and region takes place deep in GMT before parsing of common and module-specific options take place.  Based on the analysis, the input file (@earth_age) is replaced with the selected file (@earth_age_01m).  Then the module does its parsing, and it will find the remote file given as input and the first thing it does is to check if we have a local copy and if not, download).

I think I simply need to pass a flag via the filename to say "do not download" since we will exit grdcut before we actually get to the read step.

**Comment by meghanrjones** [comment]
Fixed by #5810
